cinema/views.py

```diff
@@ -1 +1,59 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import Actor, Movie, Genre, CinemaHall, MovieSession
+from cinema.serializers import (
+    ActorSerializer,
+    MovieSerializer,
+    GenreSerializer,
+    CinemaHallSerializer,
+    MovieSessionSerializer,
+    MovieListSerializer,
+    MovieSessionListSerializer,
+    MovieSessionDetailSerializer,
+    MovieDetailSerializer
+)
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieDetailSerializer
+        return MovieSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset.prefetch_related("genres", "actors")
+        return queryset
```

Comment:
What is the purpose of overriding method when you can do it in queryset attribute? Also are you sure you should prefetch fields for all actions? Fix everywhere